src/internal.c

```diff
@@ -14347,6 +14347,24 @@ static int DoHandShakeMsgType(WOLFSSL* ssl, byte* input, word32* inOutIdx,
     }
 #endif
 
+    if (ssl->options.side == WOLFSSL_CLIENT_END) {
+        switch (type) {
+        case certificate:
+        case server_key_exchange:
+        case certificate_request:
+        case server_hello_done:
```

Comment:
Is it safe to always move the below logic? With this change only client messages of the listed type will signal not resuming and peer auth not good? Does it make sense to only enable this check for `WOLFSSL_WPAS` or maybe even adding a new macro for WPA DPP? Can we leave the logic below also?

Reply:
By definition when we receive any of the listed messages as the client we are not resuming. Looking at the TLS 1.2 RFC entering this switch case is an error and we should error out here. The EAP-FAST RFC allows this behaviour so we just need to mark this connection as not resuming.

So if we enter this switch case we should at least mark it as not resuming and at worst error out. I would gate this on `WOLFSSL_WPAS` since this is not related to DPP but EAP-FAST.

The relevant sections of the RFCs are:

https://www.rfc-editor.org/rfc/rfc5246
```
   The client sends a ClientHello using the Session ID of the session to
   be resumed.  The server then checks its session cache for a match.
   If a match is found, and the server is willing to re-establish the
   connection under the specified session state, it will send a
   ServerHello with the same Session ID value.  At this point, both
   client and server MUST send ChangeCipherSpec messages and proceed
   directly to Finished messages.
```

https://datatracker.ietf.org/doc/html/rfc4851
```
   If session resumption based on server-side or client-side state
   fails, the server can gracefully fall back to a full TLS handshake.
   If the ServerHello received by the peer contains a empty Session ID
   or a Session ID that is different than in the ClientHello, the server
   may be falling back to a full handshake.  The peer can distinguish
   the server's intent of negotiating full or abbreviated TLS handshake
   by checking the next TLS handshake messages in the server response to
   the ClientHello.
```

Reply:
Fixed to error out in last commit.